luna/featurevis/transformations.py

```diff
@@ -3,53 +3,118 @@
 """
 import random
 import tensorflow as tf
+import tensorflow_addons as tfa
 
 
-def add_noise(img, noise):
+def noise(img, add_noise):
     """Adds noise to the image to be manipulated.
 
     Args:
         img (list): the image data to which noise should be added.
-        noise (boolean): whether noise should be added at all.
+        add_noise (boolean): whether noise should be added at all.
     Returns:
         list: the modified image.
     """
-    if noise:
+    if add_noise:
         if tf.compat.v1.keras.backend.image_data_format() == "channels_first":
-            img_noise = tf.random.uniform((1, 3, img.shape[2], len(img[3])),
-                                          dtype=tf.dtypes.float32)
+            img_noise = tf.random.uniform(
+                (1, 3, img.shape[2], len(img[3])), dtype=tf.dtypes.float32
+            )
         else:
-            img_noise = tf.random.uniform((img.shape),
-                                          dtype=tf.dtypes.float32)
+            img_noise = tf.random.uniform((img.shape), dtype=tf.dtypes.float32)
         img_noise = (img_noise - 0.5) * 0.25 * random.random()
         img = img + img_noise
         img = tf.clip_by_value(img, -1, 1)
     return img
 
 
-def blur_image(img, blur):
+def pad(img, pad_size, pad_mode="REFLECT", constant_value=0.5):
+    """Pad the image
+
+    Args:
+        img (list): the image data to be padded.
+        pad_size (int): size of pad.
+        mode (str, optional): mode for padding; can be "CONSTANT", "REFLECT" or "SYMMETRIC".
+        constant_value (float, optional): The scalar pad value. Defaults to 0.5.
+
+    Returns:
+        list: the padded image.
+    """
+    img = tf.pad(
+        img,
+        [(0, 0), (pad_size, pad_size), (pad_size, pad_size), (0, 0)],
+        pad_mode,
+        constant_value,
+    )
+    return img
+
+
+def jitter(img, jitter_dist, seed=None):
+    """Jitter the image & add an artistic effect to the image.
+
+    Args:
+        img (list): the image data to be jittered.
+        jitter_dist (int): the distance of effective neighborhoods.
+        seed (int, optional): Defaults to None.
+
+    Returns:
+        list: the jittered image.
+    """
+    img = tf.convert_to_tensor(img)
+    img_shape = tf.shape(img)
+    crop_shape = tf.concat(
+        [img_shape[:-3], img_shape[-3:-1] - jitter_dist, img_shape[-1:]], 0
+    )
+    crop = tf.image.random_crop(img, crop_shape, seed=seed)
+    img_shape_list = img.get_shape().as_list()
+    mid_shp_changed = [
+        img_shape_list[-3] - jitter_dist if img_shape_list[-3] is not None else None,
+        img_shape_list[-2] - jitter_dist if img_shape_list[-3] is not None else None,
+    ]
+    crop.set_shape(img_shape_list[:-3] + mid_shp_changed + img_shape_list[-1:])
+
+    return crop
+
+
+def blur(img, add_blur):
```

Comment:
Same, why do we need the second param? The other functions above do not have that kind of a switch.

Reply:
Resolved by giving transformations as function.